### PR TITLE
Deprecate proj3d.mod.

### PR DESCRIPTION
**Pull request by anntzer**
Having a function named mod that doesn't do % but computes the norm of a
vector is just a good way to trip the reader (yes, I know about the term
"modulus of a vector", but still).

Also vectorize a normals calculation.

Also we can use % instead of np.mod elsewhere.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/mpl_toolkits/mplot3d/axes3d.py [hunk 1]
```diff
@@ -1768,9 +1768,9 @@ def _shade_colors(self, color, normals, lightsource=None):
             # chosen for backwards-compatibility
             lightsource = LightSource(azdeg=225, altdeg=19.4712)
 
-        shade = np.array([np.dot(n / proj3d.mod(n), lightsource.direction)
-                          if proj3d.mod(n) else np.nan
-                          for n in normals])
+        with np.errstate(invalid="ignore"):
+            shade = ((normals / np.linalg.norm(normals, axis=1, keepdims=True))
+                     @ lightsource.direction)
```
Comment (jklymak) [inline lib/mpl_toolkits/mplot3d/axes3d.py]: OK, I couldn't even google this one.  What does the `@` operator do?

Comment (timhoffm) [reply lib/mpl_toolkits/mplot3d/axes3d.py]: That‘s a new operator introduced in python 3. Numpy uses it for matrix multiplication. `a @ b` is the same as `np.dot(a, b)`, just more readable.

**Comment by anntzer** [comment]
I can rebase this on top of #13030 once it is merged, no problem.

**Comment by jklymak** [comment]
Confused if this should be merged given #13030, of if #13030 will be merged.  If its not a problem, anyone can merge.

**Comment by QuLogic** [comment]
@anntzer agreed to wait, above.

**Comment by anntzer** [comment]
rebased and removed the deprecation notice, as #13030 already takes care of that

**Pull request merged**